Adding Vulnerability Assessment APIs on managed instance

yaakoviyun opened the pull request:
Adding Vulnerability Assessment APIs on managed instance

Those APIs are the same as the Vulnerability Assessment APIs on regular Azure db with the difference of using "/managedInstances/{managed instance name}/" instead of "/servers/{server name}/"

The Vulnerability Assessment APIs on regular Azure db are approved and can be found at:

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessments.json

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/databaseVulnerabilityAssessmentScans.json

yaakoviyun committed:
Adding Vulnerability Assessment APIs on managed instance

Adding Vulnerability Assessment APIs on managed instance

jaredmoo commented:
The linter validation failed with a few errors.

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/379065871#L761

```
{
  "Channel": "fatal",
  "Text": "Error: '$.definitions.DatabaseVulnerabilityAssessmentScanExportProperties.description' has incompatible values (---\nProperties of the export operation’s result.\n, ---\nProperties of the export operation's result.\n)."
}
(node:6908) UnhandledPromiseRejectionWarning: Error: '$.definitions.DatabaseVulnerabilityAssessmentScanExportProperties.description' has incompatible values (---
Properties of the export operation’s result.
, ---
Properties of the export operation's result.
).
```

Very subtle difference but the first description has a `’` character, second has `'` character.

yaakoviyun committed:
fixed scan operation id

AutorestCI commented:
# Automation for azure-sdk-for-go

AutorestCI commented:
# Automation for azure-sdk-for-node

AutorestCI commented:
# Automation for azure-libraries-for-java

AutorestCI commented:
# Automation for azure-sdk-for-python

yaakoviyun committed:
character mismatch fix

yaakoviyun committed:
Added needed changes to Swagger and exampels

Added needed changes to Swagger
https://github.com/Azure/azure-sdk-for-net/pull/4327#pullrequestreview-121254698

jaredmoo commented:
One last model warning: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/381535630#L768

I am waiting to approve until the API is ready in production. @ravbhatnagar , please approve design.

yaakoviyun commented:
@jhendrixMSFT - yes.

And the API is now exposed so we can merge it

jhendrixMSFT commented:
@yaakoviyun great I'll wrap up my review shortly.  There are a few model validation errors, could you have a quick look [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/381535630).

jhendrixMSFT commented:
Ping @yaakoviyun there is just one model validation failure, it's in an example.  Not blocking but would be great to have this clean before merging.

jaredmoo commented:
Is there a possibility in the future of adding `master` baselines? I want to avoid having a similar breaking change in MI like we did in DB when `master` was added.

yaakoviyun commented:
The APIs are not yet ready for exposure.

This got a low priority tag now.
I'm closing the PR and will send a new one once we will fix the APIs.

AutorestCI commented:
# Automation for azure-sdk-for-ruby
